dotnet-dependency-injection/ChangeServiceLifetimeOfAlreadyRegisteredService/ChangeServiceLifetimeOfAlreadyRegisteredService/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace ChangeServiceLifetimeOfAlreadyRegisteredService;
+
+public static class ServiceCollectionExtensions
+{
+    public static IServiceCollection ReplaceWithSingleton<TService, TImplementation>(this IServiceCollection services)
+        where TService : class
+        where TImplementation : class, TService
+        => services.Replace(new ServiceDescriptor(typeof(TService), typeof(TImplementation), ServiceLifetime.Singleton));
+
+    public static IServiceCollection ReplaceWithSingleton<TImplementation>(this IServiceCollection services)
+       where TImplementation : class
+       => services.Replace(new ServiceDescriptor(typeof(TImplementation), typeof(TImplementation), ServiceLifetime.Singleton));
```

Comment:
You could re-use the method from 7 my using `TImplementation` for both parameters. You can also pass the `ServiceLifeTime `enum into this method so you could re-use it for every method or keep it just as one. Up to you :-)